components/learn/modules/ROOT/pages/public-staging.adoc

```diff
@@ -1,39 +1,183 @@
+
+Head over to Infura, or your public node provider of your choice, create an account, and jot down your assigned project ID - we will use it later to connect to the network.
+
+=== Creating a New Account
+
+To send transactions in a testnet, you will need a new account. While you can use any method you want to create it, here we will use the `mnemonics` package, which will output a fresh mnemonic to derive a new set of accounts:
+
+[source,console]
+----
+
+$ npx mnemonics
+mistake pelican mention moment ...
+----
+
+WARNING: Make sure to keep your mnemonic secure. Even if it is just for testing purposes, there are still malicious users out there who will wreak havoc on your testnet deployment just for fun!
+
+=== Configuring the Network
+
+As we are using public nodes, we will need to sign all our transactions locally. We will use the `@truffle/hdwallet-provider` to do this, setting it up with our new mnemonic, which will be used to derive our testnet accounts. We will also need to tell the provider how to connect to the test network: we will use the Infura endpoint for this.
+
+NOTE: This part assumes you have already set up a new OpenZeppelin project. If you haven't, run `oz init` or head over to the guide on xref:deploy-and-interact.adoc#getting-started-with-the-cli[getting started with the CLI].
+
+Let's start by installing the provider:
+
+[source,console]
+----
+$ npm install --save-dev @truffle/hdwallet-provider
+----
+
+And then update our `networks.js` file with a new connection to the testnet. Here we will use Rinkeby, but you can use whichever you want:
+
+[source,js]
+----
+// network.js
+const { projectId, mnemonic } = require('./secrets.json');
+const HDWalletProvider = require('@truffle/hdwallet-provider');
+
+module.exports = {
+  networks: {
+    development: { ... },
+    rinkeby: {
+      provider: () => new HDWalletProvider(mnemonic, `https://rinkeby.infura.io/v3/${projectId}`),
+      networkId: 4,
+      gasPrice: 10e9
+    }
+  },
+};
+----
+
+NOTE: The `HDWalletProvider` can be configured to generate more than a single account from the mnemonic. Check out its https://github.com/trufflesuite/truffle/tree/master/packages/hdwallet-provider[documentation] for more info.
+
+Note in the first line that we are loading the project id and mnemonic from a `secrets.json` file, which should look like the following, but using your own values. Make sure to `.gitignore` it!
+
+[source,json]
+----
+{ 
+  "mnemonic": "mistake pelican mention moment ...", 
+  "projectId": "305c13705054a8d918ad77549e402c72"
+}
+----
+
+NOTE: Instead of a `secrets.json` file, you can use whatever secret-management solution you like for your project. A popular and simple option is to use https://github.com/motdotla/dotenv[`dotenv`] for injecting secrets as environment variables.
```

Comment:
```suggestion
TIP: Instead of a `secrets.json` file, you can use whatever secret-management solution you like for your project. A popular and simple option is to use https://github.com/motdotla/dotenv[`dotenv`] for injecting secrets as environment variables.
```